chunk/chunk_store.go

```diff
@@ -104,47 +125,132 @@ func init() {
 type Store interface {
 	Put(ctx context.Context, chunks []Chunk) error
 	Get(ctx context.Context, from, through model.Time, matchers ...*metric.LabelMatcher) ([]Chunk, error)
+	Stop()
 }
 
 // StoreConfig specifies config for a ChunkStore
 type StoreConfig struct {
 	S3URL       string
 	DynamoDBURL string
 	ChunkCache  *Cache
+
+	// Not exported as only used by tests to inject mocks
+	dynamodb dynamodbClient
+	s3       s3Client
```

Comment:
It would be better to have a separate function for creating an `AWSStore` that takes a `dynamodbClient`, `s3client`, and chunk `Cache` directly. Separates the concern of parsing & validating config from constructing a usable store.